pynitrokey/cli/start.py

```diff
@@ -93,14 +95,81 @@ def rng(count, raw, quiet):
                 print(challenge.hex())
 
 
+class CardRemovedGPGAgentException(RuntimeWarning):
+    pass
+
+
+def gpg_agent_set_identity(identity: int):
+    from pexpect import run
+
+    cmd = f"gpg-connect-agent 'SCD APDU 00 85 00 0{identity}' /bye"
+    app = run(cmd)
+    print(cmd)
+    print(app)
+    if b"ERR 100663406 Card removed" in app:
+        raise CardRemovedGPGAgentException("Card removed")
+
+
+def pcsc_set_identity(identity):
+    try:
+        from smartcard import System
+        from smartcard.CardConnection import CardConnection
+        from smartcard.Exceptions import NoCardException
+
+        def find_smartcard(uuid: typing.Optional[int] = None) -> CardConnection:
```

Comment:
I think this one returns `None` if no card is found.  That case should be handled by the caller and the type annotation should be changed.